react/vite.config.ts

```diff
@@ -1,11 +1,94 @@
-import { defineConfig } from 'vite';
+import { defineConfig, loadEnv } from 'vite';
 import react from '@vitejs/plugin-react';
 
+function getApiUrl(backend: string): string {
+  if (backend === 'development') {
+    return 'http://localhost:8888';
+  } else if (backend === 'staging') {
+    return 'https://agave.designsafe-ci.org/geo-staging/v2';
+  } else if (backend === 'production') {
+    return 'https://agave.designsafe-ci.org/geo/v2';
+  } else {
+    throw new Error('Unsupported Type');
+  }
+}
+
+function getPortalUrl(backend: string): string {
```

Comment:
rename to something more explicit like getDesignSafePortalUrl?

maybe add comment?